src/parser/iProveVisitor.js

```diff
@@ -44,7 +44,16 @@ class iProveVisitor extends ParseTreeVisitor {
           }
         }
       }
+
+      // if you havent errored and this use of an existing relation with different params is not in list, add it
+      if (!this.errors) {
+        const existing_use = this.relations.find((r) => JSON.stringify(r) === JSON.stringify({name:rel_name, numParam:params.length, params}))
```

Comment:
I think it's a lot nicer to use `_.isEqual(r, {name:rel_name, numParam:params.length, params})` instead of `JSON.stringify` which has a big overhead